spec/rubocop/cop/rspec/capybara/specific_matcher_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::Capybara::SpecificMatcher do
+  it 'does not register an offense for abstract matcher when elements with' \
+    'no matcher specified in detail' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_selector('article')
+      expect(page).to have_none_of_selectors('body')
+      expect(page).to have_css('tbody')
+    RUBY
+  end
+
+  it 'does not register an offense for abstract matcher when elements with' \
+    'not element' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_no_css('.a')
+      expect(page).to has_css?('#button')
+      expect(page).to has_no_css?('[table]')
+    RUBY
+  end
+
+  it 'registers an offense when using `#have_selector`' do
+    expect_offense(<<-RUBY)
+    expect(page).to have_selector('button')
+                    ^^^^^^^^^^^^^^^^^^^^^^^ Use `#have_button` instead of `#have_selector`.
```

Comment:
There is no ambiguity to the subject of the offence. We point at the method call. Is it necessary to tell the method name in the offence message again?

Reply:
I would have preferred more descriptive, but it certainly seems redundant in this case.